pennylane/measurements/mid_measure.py

```diff
@@ -123,13 +136,14 @@ class MeasurementValue(Generic[T]):
     Measurements on a single qubit in the computational basis are assumed.
 
     Args:
-        measurement_ids (list[str]): The id of the measurement that this object depends on.
+        measurements (list[.MidMeasureMP]): The measurement(s) that this object depends on.
         processing_fn (callable): A lazily transformation applied to the measurement values.
     """
 
-    def __init__(self, measurement_ids, processing_fn):
-        self.measurement_ids = measurement_ids
+    def __init__(self, measurements, processing_fn):
+        self.measurements = measurements
         self.processing_fn = processing_fn
+        self.measurement_ids = [m.id for m in measurements]
```

Comment:
Is keeping `self.measurement_ids` necessary? It seems like almost everywhere this is used, it's just `len(self.measurement_ids)`, which could just as well be `len(self.measurements)`. Exceptions are in `_merge`, where it looks to me like you could still use `self.measurements` instead with a small tweak, and in `__str__`, where I guess you would still want to make this list, but could do it in the function. 

I guess this isn't of much consequence, but it just seems like it could be confusing in the future to figure out why both measurements and measurement_ids are here.